Data_Structures_Questions.md

```diff
@@ -4,50 +4,94 @@ Answer the following questions for each of the data structures you implemented a
 
 1. What is the runtime complexity of `enqueue`?
 
+Constant, `O(1)`.
+
 2. What is the runtime complexity of `dequeue`?
 
+Constant, `O(1)`.
+
 3. What is the runtime complexity of `len`?
 
+Constant, `O(1)`.
+
 ## Binary Search Tree
 
 1. What is the runtime complexity of `insert`? 
 
+Linear based on the size of the BST, `O(k)`.
+
 2. What is the runtime complexity of `contains`?
 
+Linear based on the size of the BST, `O(k)`.
+
 3. What is the runtime complexity of `get_max`? 
 
+Linear based on the size of the BST, `O(k)`.
+
```

Comment:
i believe the methods for bst are O(log n), we are traversing the levels instead of inspecting each element, we continue traversing left or right. for the get_max, even though we need to only go right, it still depends on the number of elements, and that contributes to how far right we need to go